### PR TITLE
Fix hoisted script propagation in content collection pages

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Replacement for https://github.com/withastro/astro/pull/6047

## Testing

- Unskipped previously failing test

## Docs

N/A

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
Can't say I grok this fully, but the test looks good and I'm cool with the approach. LGTM!

File: packages/astro/src/core/build/types.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ export interface PageBuildData {
 	moduleSpecifier: string;
 	css: Map<string, { depth: number; order: number }>;
 	contentCollectionCss: Map<string, Set<string>>;
```
Comment (bholmesdev) [inline packages/astro/src/core/build/types.ts:24]: nit: Meant to rename `contentCollectionCss` to `propagatedCss` or `propagatedStyles` for consistency. Possible to add to this PR?

File: packages/astro/src/core/build/graph.ts [hunk 2]
```diff
@@ -40,9 +40,10 @@ export function moduleIsTopLevelPage(info: ModuleInfo): boolean {
 // This could be a .astro page, a .markdown or a .md (or really any file extension for markdown files) page.
 export function* getTopLevelPages(
 	id: string,
-	ctx: { getModuleInfo: GetModuleInfo }
+	ctx: { getModuleInfo: GetModuleInfo },
+	walkUntil?: (importer: string) => boolean
 ): Generator<[ModuleInfo, number, number], void, unknown> {
-	for (const res of walkParentInfos(id, ctx)) {
+	for (const res of walkParentInfos(id, ctx, walkUntil)) {
```
Comment (bholmesdev) [inline packages/astro/src/core/build/graph.ts:46]: This refactor is unused looks like. Good to revert changes to this file!

File: packages/astro/src/content/vite-plugin-content-assets.ts [hunk 3]
```diff
@@ -73,42 +76,95 @@ export function astroContentAssetPropagationPlugin({ mode }: { mode: string }):
...
+
+						let newCode = chunk.code;
+						if (entryCSS.size) {
+							newCode = newCode.replace(
+								JSON.stringify(LINKS_PLACEHOLDER),
+								JSON.stringify([...entryCSS])
+							);
+						}
+						if (entryScripts.size) {
+							const entryFileNames = new Set<string>();
```
Comment (bholmesdev) [inline packages/astro/src/content/vite-plugin-content-assets.ts:137]: _Love_ that we can look these up now 👏 Way simpler than scary naming conventions!

**Review by bholmesdev — COMMENTED** [review]
Couple comments, mostly for cleaning up code I wrote originally. Propagreat stuff!

**Pull request merged**